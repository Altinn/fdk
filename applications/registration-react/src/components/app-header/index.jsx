import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';

import {
  fetchUserIfNeeded
} from '../../actions/index';
import localization from '../../utils/localization';
import './index.scss';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
    this.props.dispatch(fetchUserIfNeeded());
  }

  toggle() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  }

  render() {
    const { userItem } = this.props;
    return (
      <header>
        <div>
          <a
            id="focus-element"
            className="uu-invisible"
            href={`${location.pathname}#content`}
            aria-hidden="true"
          >
            Hopp til hovedinnhold
          </a>
        </div>
        <div id="skip-link-wrap">
          <a id="skip-link" href={`${location.pathname}#content`}>Hopp til hovedinnhold</a>
        </div>
        <div className="fdk-header-beta">
          {localization.beta.header}
          <br className="d-md-none" />
          {localization.beta.first}
          <a className="white-link" href="mailto:fellesdatakatalog@brreg.no">{localization.beta.second}</a> {localization.beta.last}
        </div>
        <div className="fdk-header">
          <div className="container">
            <div className="row">
              <div className="col-6 col-md-4">
                <a
                  title="Link til Felles datakatalog"
                  href="/"
                >
                  <span className="uu-invisible" aria-hidden="false">GÃ¥ til forside</span>
                  <img className="fdk-logo" src="/static/img/fdk-logo@2x.png" alt="Logo for Felles datakatalog" />
                </a>
              </div>

              <div className="col-6 col-md-4 d-flex justify-content-center align-items-center">
                <span><strong>Registrering</strong></span>
              </div>
              <div className="col-md-4 d-flex align-items-center fdk-header-text_items justify-content-end">
                {userItem && userItem.name &&
                  <div className="fdk-margin-right-double">
                    <i className="fa fa-user fdk-fa-left fdk-color-cta3" />
                    {userItem.name}
                  </div>
                }
                <div className="fdk-margin-right-double fdk-auth-link">
                  <a href={`${window.location.origin  }/logout`}>{localization.app.logOut}</a>
                </div>
                <div>
                  <ButtonDropdown className="btn-group-default" isOpen={this.state.dropdownOpen} toggle={this.toggle}>
                    <DropdownToggle className="fdk-button fdk-button-default fdk-button-menu dropdown-toggle btn-default" caret>
                      <i className="fa fa-bars fdk-fa-dark fdk-fa-left" />
                      <span>Meny</span>
                    </DropdownToggle>
                    <DropdownMenu>
                      <DropdownItem>Veileder</DropdownItem>
                      <DropdownItem>Standard</DropdownItem>
                      <DropdownItem>
                        <a href="">Felles datakatalog</a>
                      </DropdownItem>
                    </DropdownMenu>
                  </ButtonDropdown>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="fdk-breadcrumb breadcrumb d-flex justify-content-between">
          <span></span>
          <span>{localization.app.autoSave}</span>
        </div>
      </header>
    );
  }
}

Header.defaultProps = {

};

Header.propTypes = {
  dispatch: PropTypes.func.isRequired
};

function mapStateToProps({ user }, ownProps) {
  const { userItem } = user || {
    userItem: null
  }

  return {
    userItem
  };
}

export default connect(mapStateToProps)(Header);
