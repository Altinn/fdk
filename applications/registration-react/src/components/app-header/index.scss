@import '../../assets/style/colors';

.fdk-header-text_items {
  font-size: 0.9rem;
  white-space: nowrap;
}

.fdk-auth-link {
  a {
    color: $color1;
    font-size: .9rem;
  }
}