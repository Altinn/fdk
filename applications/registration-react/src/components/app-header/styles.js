import 'designsystemet/fdk-designsystem-bootstrap4/scss/helper.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/register.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/typo.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/common.scss';
import 'designsystemet/fdk-designsystem-bootstrap4/scss/animations.scss';
import 'font-awesome/scss/font-awesome.scss';
