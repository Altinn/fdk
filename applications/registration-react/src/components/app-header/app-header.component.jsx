import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

import { fetchUserIfNeeded } from '../../actions/index';
import localization from '../../utils/localization';
import '../../assets/style/bootstrap-override.scss';
import './styles';
import './app-header.scss';

export class Header extends React.Component {
  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
    this.props.dispatch(fetchUserIfNeeded());
  }

  toggle() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  }

  render() {
    let isApiReg;
    let isDatasetReg;
    if (this.props.location) {
      isApiReg = this.props.location.pathname.split('/')[3] === 'apis';
      isDatasetReg = this.props.location.pathname.split('/')[3] === 'datasets';
    }
    const { userItem } = this.props;
    return (
      <header>
        <div>
          <a
            id="focus-element"
            className="uu-invisible"
            href={`${location.pathname}#content`}
            aria-hidden="true"
          >
            {localization.app.skipLink}
          </a>
        </div>
        <div id="skip-link-wrap">
          <a id="skip-link" href={`${location.pathname}#content`}>
            {localization.app.skipLink}
          </a>
        </div>
        <div className="fdk-header">
          <div className="container">
            <div className="row">
              <div className="col-6 col-md-4">
                <a title="Link til Felles datakatalog" href="/">
                  <span className="uu-invisible" aria-hidden="false">
                    {localization.app.navigateFrontpage}
                  </span>
                  <img
                    className="fdk-logo"
                    src="/static/img/logo-registrering.svg"
                    alt="Logo for Felles datakatalog"
                  />
                </a>
              </div>

              <div className="col-6 col-md-4 d-flex justify-content-center align-items-center">
                <span>
                  <strong>
                    {isApiReg && localization.header["Registration of API's"]}
                    {isDatasetReg &&
                      localization.header['Registration of Datasets']}
                    {!isDatasetReg && !isApiReg && localization.app.title}
                  </strong>
                </span>
              </div>
              <div className="col-md-4 d-flex align-items-center fdk-header-text_items justify-content-end">
                {userItem &&
                  userItem.name && (
                    <div className="mr-4">
                      <i className="fa fa-user fdk-fa-left fdk-color-cta3" />
                      {userItem.name}
                    </div>
                  )}
                {userItem && (
                  <div className="mr-4 fdk-auth-link">
                    <a href={`${window.location.origin}/logout`}>
                      {localization.app.logOut}
                    </a>
                  </div>
                )}
                {!userItem && (
                  <div className="mr-4 fdk-auth-link">
                    <a href={`${window.location.origin}/login`}>
                      {localization.app.logIn}
                    </a>
                  </div>
                )}
              </div>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

Header.defaultProps = {
  userItem: null
};

Header.propTypes = {
  dispatch: PropTypes.func.isRequired,
  userItem: PropTypes.object,
  location: PropTypes.object.isRequired
};

function mapStateToProps(props) {
  const { user } = props;
  const { userItem } = user || {
    userItem: null
  };

  return {
    userItem
  };
}

export default withRouter(connect(mapStateToProps)(Header));
