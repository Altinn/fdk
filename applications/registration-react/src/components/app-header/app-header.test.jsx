import React from 'react';
import { shallow } from 'enzyme';
import { AppHeaderPure } from './app-header.component';

let defaultProps;
let wrapper;
let dispatch;

beforeEach(() => {
  dispatch = jest.fn();
  defaultProps = {
    dispatch,
    location: { pathname: 'abc' }
  };
  wrapper = shallow(<AppHeaderPure {...defaultProps} />);
});

test('should render AppHeader correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
