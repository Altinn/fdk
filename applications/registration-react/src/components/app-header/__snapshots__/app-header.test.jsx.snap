// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppHeader correctly 1`] = `
<header>
  <div>
    <a
      aria-hidden="true"
      className="uu-invisible"
      href="abc#content"
      id="focus-element"
    >
      Hopp til hovedinnhold
    </a>
  </div>
  <div
    id="skip-link-wrap"
  >
    <a
      href="abc#content"
      id="skip-link"
    >
      Hopp til hovedinnhold
    </a>
  </div>
  <div
    className="fdk-header"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-6 col-md-4"
        >
          <a
            href="/"
            title="Link til Felles datakatalog"
          >
            <span
              aria-hidden="false"
              className="uu-invisible"
            >
              GÃ¥ til forside
            </span>
            <img
              alt="Logo for Felles datakatalog"
              className="fdk-logo"
              src="/static/img/logo-registrering.svg"
            />
          </a>
        </div>
        <div
          className="col-md-4 d-flex align-items-center fdk-header-text_items justify-content-end"
        >
          <div
            className="mr-4 fdk-auth-link"
          >
            <button
              className="fdk-btn-no-border"
              onClick={[Function]}
              type="button"
            >
              Logg inn
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
